gif/Form1.cs

```diff
@@ -131,15 +138,33 @@ private void findInfectedFiles()
                     progressBar.Invoke(new MethodInvoker(delegate { progressBar.Value += 1; }));
                 }
 
-                string[] lines = File.ReadAllLines(file);
-                string? firstOccurance = lines.FirstOrDefault(line => line.Contains("I hope you realize you aren't safe in workshop anymore"));
-
-                if (firstOccurance != null)
+                try
                 {
-                    if (lstInfectedFiles.InvokeRequired)
+                    bool firstOccurance = false;
+                    IEnumerable<string> lines = File.ReadLines(file);
+                    foreach (string line in lines)
                     {
-                        lstInfectedFiles.Invoke(new MethodInvoker(delegate { lstInfectedFiles.Items.Add(file); }));
+                        if (line.Contains("I hope you realize you aren't safe in workshop anymore"))
+                            firstOccurance = true;
                     }
+
+                    if (firstOccurance)
```

Comment:
I don't see a reason for this bool and the check for it to be true, it's not used anywhere else. You can condense it so it will just add it to the Listbox on the if statement for the line.contains(string) and then break out of the foreach loop and scan the next file which will in turn also prevent it from parsing the file further and save time.

Reply:
That bool was added in your code so I kept that there but I'll look into that